javacord-core/src/main/java/org/javacord/core/util/handler/channel/thread/ThreadUpdateHandler.java

```diff
@@ -52,22 +80,155 @@ private void handleThread(final JsonNode jsonChannel) {
 
         //Handling whether the name has changed
         final String oldName = thread.getName();
         final String newName = jsonChannel.get("name").asText();
         if (!Objects.deepEquals(oldName, newName)) {
             thread.setName(newName);
+
             final ServerChannelChangeNameEvent event =
                     new ServerChannelChangeNameEventImpl(thread, newName, oldName);
 
-            if (server.isReady()) {
-                api.getEventDispatcher().dispatchServerChannelChangeNameEvent(
-                        (DispatchQueueSelector) thread.getServer(), thread.getServer(), thread, event);
+            api.getEventDispatcher().dispatchServerChannelChangeNameEvent(
+                    (DispatchQueueSelector) thread.getServer(), thread.getServer(), thread, event);
+        }
+
+        final int oldMessageCount = thread.getMessageCount();
+        final int newMessageCount = jsonChannel.get("message_count").asInt();
+        if (oldMessageCount != newMessageCount) {
+            thread.setMessageCount(newMessageCount);
+
+            final ServerThreadChannelChangeMessageCountEvent event =
+                    new ServerThreadChannelChangeMessageCountEventImpl(thread, newMessageCount, oldMessageCount);
+
+            api.getEventDispatcher().dispatchServerThreadChannelChangeMessageCountEvent(
+                    (DispatchQueueSelector) thread.getServer(), thread.getServer(), thread, event);
+        }
+
+        final int oldMemberCount = thread.getMemberCount();
+        final int newMemberCount = jsonChannel.get("member_count").asInt();
+        if (oldMemberCount != newMemberCount) {
+            thread.setMemberCount(newMemberCount);
+
+            final ServerThreadChannelChangeMemberCountEvent event =
+                    new ServerThreadChannelChangeMemberCountEventImpl(thread, newMemberCount, oldMemberCount);
+
+            api.getEventDispatcher().dispatchServerThreadChannelChangeMemberCountEvent(
+                    (DispatchQueueSelector) thread.getServer(), thread.getServer(), thread, event);
+        }
+
+        final Set<ThreadMember> oldMembers = thread.getMembers();
+        final Set<ThreadMember> newMembers = new HashSet<>();
+        if (jsonChannel.hasNonNull("member")) {
+            for (final JsonNode jsonMember : jsonChannel.get("member")) {
+                final ThreadMember member = new ThreadMemberImpl(api, server, jsonMember);
+                newMembers.add(member);
             }
         }
 
+        if (!Objects.deepEquals(oldMembers, newMembers)) {
+            thread.setMembers(newMembers);
+
+            final ServerPrivateThreadJoinEvent event =
+                    new ServerPrivateThreadJoinEventImpl(thread, newMembers,
+                            oldMembers);
+
+            api.getEventDispatcher().dispatchServerPrivateThreadJoinEvent(
+                    (DispatchQueueSelector) thread.getServer(), thread.getServer(), thread, event);
+        }
```

Comment:
`member` is not an array as I have shown you.
```suggestion
        if (jsonChannel.hasNonNull("member")) {
            final Set<ThreadMember> members = thread.getMembers();
            final ThreadMember member = new ThreadMemberImpl(api, server, jsonChannel.get("member"));
            members.add(member);
            thread.setMembers(newMembers);

            final ServerPrivateThreadJoinEvent event =
                    new ServerPrivateThreadJoinEventImpl(thread, newMembers,
                            oldMembers);

            api.getEventDispatcher().dispatchServerPrivateThreadJoinEvent(
                    (DispatchQueueSelector) thread.getServer(), thread.getServer(), thread, event);
        }
```
I don't have access to private threads to test it but this should work. The data to dispatch for this event should probably be just just the ServerThreadChannel. And then it's probably also the best to move this code to the very end so it gets executed last

Reply:
> The data to dispatch for this event should probably be just just the ServerThreadChannel

Any reason you have not applied this requested change and exchanged the old/new members with the ServerThreadChannel? I am not sure if you even have to do this. It may be possible that it's already possible to get the channel by default through this event because it's already extending something.